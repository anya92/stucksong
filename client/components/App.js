import React, { Component } from 'react';
import { connect } from 'react-redux';
import { ThemeProvider } from 'glamorous';
import {
  oneOfType,
  bool,
  object,
  func,
} from 'prop-types';

import { theme, Container } from '../styles/global';
import { fetchUser } from '../actions';

import Routes from './Routes';
import ErrorBoundry from './Error/ErrorBoundry';

function mapStateToProps(state) {
  return {
    auth: state.auth,
  };
}

export class App extends Component {
  static propTypes = {
    auth: oneOfType([bool, object]),
    fetchUser: func.isRequired,
  }

  static defaultProps = {
    auth: null,
  }

  componentDidMount() {
    this.props.fetchUser();
  }

  render() {
    const { auth } = this.props;
    if (auth === null) return <div />;
    return (
      <ThemeProvider theme={theme}>
        <ErrorBoundry>
          <Container>
            <Routes auth={auth} />
          </Container>
        </ErrorBoundry>
      </ThemeProvider>
    );
  }
}

export default connect(mapStateToProps, { fetchUser })(App);
