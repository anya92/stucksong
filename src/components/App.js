import React, { Component } from 'react';
import {
	BrowserRouter as Router,
	Route
} from 'react-router-dom';
import { connect } from 'react-redux';

import { fetchUser } from '../actions';

import requireAuth from './require_authentication';
import redirectToTracks from './redirect_to_tracks';
import Navbar from './Navbar';
import Home from './Home';
import Tracks from './Tracks';
import Artists from './Artists';
import Recently from './Recently';
import CreatePlaylist from './CreatePlaylist';

class App extends Component {
	componentDidMount() {
		this.props.fetchUser();
	}

	render() {
		return this.props.auth === null ? <div className="loading"></div> : (
			<div>
				<Router>
					<div>
						<Route component={Navbar} />
						<Route exact path='/' component={redirectToTracks(Home)} />
						<Route path='/top-tracks' component={requireAuth(Tracks)} />
						<Route path='/top-artists' component={requireAuth(Artists)} />
						<Route path='/recently-played' component={requireAuth(Recently)} />
						<Route path='/create-playlist' component={requireAuth(CreatePlaylist)} />
					</div>
				</Router>
			</div>
		);
	}
}

function mapStateToProps(state) {
	return {
		auth: state.auth
	};
}

export default connect(mapStateToProps, { fetchUser })(App);
