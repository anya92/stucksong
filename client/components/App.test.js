import React from 'react';
import { shallow } from 'enzyme';

import { App } from './App';

describe('App component', () => {
  let wrapper;
  let fetchUser;
  beforeEach(() => {
    fetchUser = jest.fn();
    const mockProps = {
      auth: null,
      fetchUser,
    };
    wrapper = shallow(<App {...mockProps} />);
  });

  it('calls fetchUser in componentDidMount', () => {
    // wrapper.instance().componentDidMount();
    expect(fetchUser).toHaveBeenCalled();
  });

  it('renders correctly', () => {
    wrapper.setProps({ auth: true });
    expect(wrapper).toMatchSnapshot();
  });
});
