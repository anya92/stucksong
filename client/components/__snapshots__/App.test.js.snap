// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<ThemeProvider
  theme={
    Object {
      "backgroundColor": "#f4f4f4",
      "boxShadow": "0 1px 2px rgba(0, 0, 0, 0.07)",
      "fontFamily": "-apple-system, BlinkMacSystemFont, \\"Nunito\\", \\"Segoe UI\\", Roboto, sans-serif",
      "gradient": "linear-gradient(to top right, #1d2b64, #f8cdda)",
      "mainColor": "#B53471",
    }
  }
>
  <ErrorBoundry>
    <glamorous(div)>
      <Routes
        auth={true}
      />
    </glamorous(div)>
  </ErrorBoundry>
</ThemeProvider>
`;
